src/lsm/groove.zig

```diff
@@ -571,15 +571,23 @@ pub fn GrooveType(
             assert(groove.prefetch_ids.count() == 0);
         }
 
-        /// This must be called by the state machine for every key to be prefetched.
-        /// We tolerate duplicate IDs enqueued by the state machine.
-        /// For example, if all unique operations require the same two dependencies.
-        pub fn prefetch_enqueue(groove: *Groove, key: PrimaryKey) void {
-            if (!has_id) {
-                groove.prefetch_ids.putAssumeCapacity(key, {});
-                return;
+        /// This function makes a best-effort attempt to ascertain the existence of the passed key
+        /// in the IdTree without any I/O. If the existence of the key cannot be ascertained without
+        /// I/O, then it is enqueued for prefetching.
+        pub fn prefetch_enqueue_negative(groove: *Groove, key: PrimaryKey) void {
+            if (!has_id) groove.prefetch_enqueue();
```

Comment:
This was discussed in the meeting today morning, I'm planning on removing the `if (!has_id) groove.prefetch_enqueue();` entirely since this routine is only to be invoked for grooves that have an ID field.

And there indeed was a bug in there, should've been `if (!has_id) return groove.prefetch_enqueue();`. :D